### ISSUE TITLE
Learner1D:  restriction on maximum precision of the x variable

### ISSUE THREAD

**Issue by SultanOrazbayev** [python-adaptive/adaptive#322]
Is there a way to control maximum precision of the x variable? For example, if I am only interested in x to 2 decimal places, the runner would not have to try values that have 3+ decimal places.

This code will keep searching for very precise answer, but I might be happy with the answer up to 2 decimal places:

```python
def f(x):
    return (x-0.335)**2

import adaptive
adaptive.notebook_extension()
learner = adaptive.Learner1D(f, bounds=(0, 1))
runner = adaptive.Runner(learner, goal=lambda l: l.loss() < 1e-6)
runner.live_plot()
```

**Comment by basnijholt** [comment]
Yes, check out this loss function https://github.com/python-adaptive/adaptive/blob/1597e9ae9bbbd5c1d1a62e16a633fc7628393d69/adaptive/learner/learner1D.py#L80

**Comment by basnijholt** [comment]
This loss-function was missing from the docs but I have just pushed a fix where I add it (https://github.com/python-adaptive/adaptive/commit/4e34aba8cf52b22d642a85a5da15d41be8fa86a3).

**Comment by SultanOrazbayev** [comment]
Thank you, this does what I want! It doesn't work the way I expected (with specific decimal precision), but the end result is the same. One potential problem though is this approach (with the interval length) sometimes settles at suboptimal results:

```python
# running this snippet multiple times in a jupyter notebook sometimes
# shows just a very small number of evaluated points (3-4 points)
# this is due to the fairly high min_length value below

def f(x):
    return (x-0.335)**2

import adaptive
adaptive.notebook_extension()
from adaptive.learner.learner1D import resolution_loss_function

loss = resolution_loss_function(min_length=0.1, max_length=1)
learner = adaptive.Learner1D(f, bounds=(-1, 1), loss_per_interval=loss)
runner = adaptive.Runner(learner, goal=lambda l: l.loss() < 1e-6)
runner.live_plot()
```

This of course is due to the high `min_length` value, so is less of an issue for small `min_length` values.

Kindly note that the docs appear to be incorrect (the suggested version is below, not sure if this is worth a pull request):

```python
    Examples
    --------
    >>> def f(x):
    ...     return x**2
    >>>
    >>> loss = resolution_loss_function(min_length=0.01, max_length=1)
    >>> learner = adaptive.Learner1D(f, bounds=(-1, -1), loss_per_interval=loss)
```

**Comment by basnijholt** [comment]
Definitely worth a PR! If you won't add it, I can do it.
